### DISCUSSION TITLE
Create pool and failed

### DISCUSSION THREAD

**Discussion by df-bingo** [rook/rook#9157] [Q&A] [answered]
kubectl apply -f pool.yaml
##############################
apiVersion: ceph.rook.io/v1
kind: CephBlockPool
metadata:
  name: replicapool  
  namespace: rook-ceph
spec:
  failureDomain: host
  replicated:
    size: 3
    requireSafeReplicaSize: true
##############################

view rook-ceph-operator log
###############################
E | ceph-block-pool-controler: failed to reconcile. invaild pool CR "replicapool" spec: failed to get crush map: failed to unmarshal crush map: json: cannot unmarshal number -1 into Go struct field stepSpec.rules.steps.num of type uint
###############################

Please help!!

**Comment by travisn** [comment, marked as answer]
A few questions:
1. Did you perform some operation on the CRUSH map to get into this state? I haven't seen this error before, just wondering how to repro.
2. Is this a new install, or you were adding a new pool?
3. What does `ceph status` show in the toolbox?
4. Does `ceph osd crush dump` succeed?

**Reply by df-bingo** [reply]
The answers are as follows：
1、Added some crush rules，and some bucket types，like datacenter、rack，etc.
      I rolled back the above operation, but the problem still exists.
2、Just installed，Pools can be added through the command line in the toolbox. Now too.
3、it shows “health ok”. Everything looks normal.
4、The command can be executed successfully.

**Reply by travisn** [reply]
It sounds like you have modified the CRUSH rules in an unexpected way for Rook. If you want to add more topology-based rules, see [this topic](https://rook.github.io/docs/rook/v1.7/ceph-cluster-crd.html#osd-topology) so you don't have to modify the CRUSH rules directly.

**Reply by df-bingo** [reply]
Thanks! I will try.
Do you mean not to add crush rules with commands through the toolbox?
